src/connection.js

```diff
@@ -1638,8 +1667,23 @@ class Connection extends EventEmitter {
       this.request.rows = [];
       this.request.rst = [];
 
+      let message;
+
+      this.request.once('cancel', () => {
+        // There's two ways to handle request cancelation:
+        if (message.writable) {
+          // - if the message is still writable, we'll set the ignore bit
+          message.ignore = true;
+        } else {
+          // - but if the message has been ended (and thus has been fully sent off),
+          //   we need to send an `ATTENTION` message to the server
```

Comment:
Can the connection already be in the SENT_ATTENTION state, because `requesttimeout()` has executed?
Or in any other state?

Reply:
You're right. 🙇 The request timeout functionality was still handling the sending of the attention message itself. I changed it to simply call `this.request.cancel` instead. This way, only the first call to `.cancel` will have any effect, and all later calls will not do anything.